server/game/cards/02_SHD/upgrades/TopTarget.ts

```diff
@@ -0,0 +1,28 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { CardType, KeywordName, RelativePlayer } from '../../../core/Constants';
+
+export default class TopTarget extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '4282425335',
+            internalName: 'top-target',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addGainKeywordTargetingAttached({
+            keyword: KeywordName.Bounty,
+            ability: {
+                title: 'Heal 4 damage from a unit or base. If the Bounty unit is unique, heal 6 damage instead.',
+                targetResolver: {
+                    controller: RelativePlayer.Any,
+                    cardTypeFilter: CardType.Base,
+                    immediateEffect: AbilityHelper.immediateEffects.heal((context) => ({ amount: context.source.unique ? 6 : 4 })),
```

Comment:
it is not context.source.parentCard.unique ? like all other upgrades ?

Reply:
In this case it's because the ability is owned by the parent card. So e.g. if Top Target is attached to Wampa, then context.source here is Wampa because the Bounty ability technically belongs to Wampa.